### ISSUE TITLE
Argument 1 passed to ContainerConfigurator must be an instance of RectorPrefix20210909\...\ContainerBuilder, instance of Symfony\...\MergeExtensionConfigurationContainerBuilder given

### ISSUE THREAD

**Issue by karser** [rectorphp/rector#6698]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | v0.11.53 (ae25b721b3193e6da82b711c2401a8c309c6b8b7)             |
| phpstan/phpstan | v0.12.98 |
| phpstan/phpstan-symfony | v0.12.44 |
| symfony | v4.4.27 |

I started getting this error after I added `console_application_loader: tests/console-application.php` to phpstan.neon.

Here is the error:
```
vendor/bin/rector process --ansi -vvv
[parsing] src/Command/CreateUserCommand.php
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to RectorPrefix20210909\Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator::__construct() must be an instance of RectorPrefix20210909\Symfony\Component\DependencyInjection\ContainerBuilder, instance of Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationContainerBuilder given, called in /vendor/symfony/dependency-injection/Loader/PhpFileLoader.php on line 67 and defined in /vendor/rector/rector/vendor/symfony/dependency-injection/Loader/Configurator/ContainerConfigurator.php:37
Stack trace:
#0 /vendor/symfony/dependency-injection/Loader/PhpFileLoader.php(67): RectorPrefix20210909\Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator->__construct()
#1 /vendor/twig/extra-bundle/DependencyInjection/TwigExtraExtension.php(32): Symfony\Component\De in /vendor/rector/rector/vendor/symfony/dependency-injection/Loader/Configurator/ContainerConfigurator.php on line 37
Fatal error: Uncaught TypeError: Argument 1 passed to RectorPrefix20210909\Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator::__construct() must be an instance of RectorPrefix20210909\Symfony\Component\DependencyInjection\ContainerBuilder, instance of Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationContainerBuilder given, called in /vendor/symfony/dependency-injection/Loader/PhpFileLoader.php on line 67 and defined in /vendor/rector/rector/vendor/symfony/dependency-injection/Loader/Configurator/ContainerConfigurator.php:37
Stack trace:
#0 /vendor/symfony/dependency-injection/Loader/PhpFileLoader.php(67): RectorPrefix20210909\Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator->__construct()
#1 /vendor/twig/extra-bundle/DependencyInjection/TwigExtraExtension.php(32): Symfony\Component\De in /vendor/rector/rector/vendor/symfony/dependency-injection/Loader/Configurator/ContainerConfigurator.php on line 37
```

## Minimal PHP Code Causing Issue

phpstan.neon (only relevant stuff)
```
includes:
    - vendor/symplify/phpstan-extensions/config/config.neon
    - vendor/phpstan/phpstan-symfony/extension.neon
parameters:
    symfony:
        container_xml_path: var/cache/dev/srcApp_KernelDevDebugContainer.xml
        # if I comment out the next line the error is gone
        console_application_loader: tests/console-application.php
```

tests/console-application.php
```
<?php

use App\Kernel;
use Symfony\Bundle\FrameworkBundle\Console\Application;

require __DIR__ . '/../config/bootstrap.php';
$kernel = new Kernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
return new Application($kernel);
```

## Expected Behaviour

`vendor/bin/rector process` completes without fatal error

**Comment by samsonasik** [comment]
Could you create sample repository that reproduce the issue? Thank you.

**Comment by dorrogeray** [comment]
Hi, I just had same issue with:

Rector version | 0.12.16
-- | --
phpstan/phpstan | 1.4.2
symfony | v5.4.2

I noticed that this happens when I manually clear the cache: `rm -rf var/cache/*`, and then call rector `vendor/bin/rector process src/Repository/SomeRepository.php --clear-cache -vvv`. The workaround is to call `bin/console` first, which forces symfony to build the DI container. After that, rector works as expected, and same call passes and fixes the issues in the file.

The project is pretty large, so time required for DI might be a factor. I will try to isolate the issue more if I have time, but I thought this could help to narrow the search.

**Comment by TomasVotruba** [comment]
The config container has to be build first, to be albe to use the dumped `*.xml`. That's the way Symfony works and we build the rule around it.

Yet, this might help with the autoloading race condition: createForm

**Comment by dorrogeray** [comment]
@TomasVotruba Thanks for the incredibly fast reaction. :+1: I have done some additional testing, and I think that the problem comes from the `phpstan/phpstan-doctrine` phpstan extension. I have done following experiment:

![image](https://user-images.githubusercontent.com/4999006/158422544-20c4dab9-dcbb-4c0e-be2e-1a5eb2a6bf62.png)

And then I have just run the rector, which raised the exception and I got this callstack:
![image](https://user-images.githubusercontent.com/4999006/158423362-b49b2606-99ce-48e8-8926-128d311db812.png)

Full callstack:
```
In MergeExtensionConfigurationPass.php line 67:
               
  [Exception]  
               

Exception trace:
  at /var/www/symfony/vendor/symfony/dependency-injection/Compiler/MergeExtensionConfigurationPass.php:67
 Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationPass->process() at /var/www/symfony/vendor/symfony/http-kernel/DependencyInjection/MergeExtensionConfigurationPass.php:42
 Symfony\Component\HttpKernel\DependencyInjection\MergeExtensionConfigurationPass->process() at /var/www/symfony/vendor/symfony/dependency-injection/Compiler/Compiler.php:82
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /var/www/symfony/vendor/symfony/dependency-injection/ContainerBuilder.php:757
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /var/www/symfony/vendor/symfony/http-kernel/Kernel.php:548
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /var/www/symfony/src/Kernel.php:105
 App\Kernel->initializeContainer() at /var/www/symfony/vendor/symfony/http-kernel/Kernel.php:789
 Symfony\Component\HttpKernel\Kernel->preBoot() at /var/www/symfony/vendor/symfony/http-kernel/Kernel.php:128
 Symfony\Component\HttpKernel\Kernel->boot() at /var/www/symfony/tests/object-manager.php:14
 require() at /var/www/symfony/vendor/phpstan/phpstan-doctrine/src/Type/Doctrine/ObjectMetadataResolver.php:77
 PHPStan\Type\Doctrine\ObjectMetadataResolver->loadObjectManager() at /var/www/symfony/vendor/phpstan/phpstan-doctrine/src/Type/Doctrine/ObjectMetadataResolver.php:56
 PHPStan\Type\Doctrine\ObjectMetadataResolver->getObjectManager() at /var/www/symfony/vendor/phpstan/phpstan-doctrine/src/Type/Doctrine/QueryBuilder/Expr/ExpressionBuilderDynamicReturnTypeExtension.php:47
 PHPStan\Type\Doctrine\QueryBuilder\Expr\ExpressionBuilderDynamicReturnTypeExtension->getTypeFromMethodCall() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:3931
 PHPStan\Analyser\MutatingScope->methodCallReturnType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1854
 PHPStan\Analyser\MutatingScope->PHPStan\Analyser\{closure}() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1860
 PHPStan\Analyser\MutatingScope->resolveType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:543
 PHPStan\Analyser\MutatingScope->getType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:813
 PHPStan\Analyser\MutatingScope->resolveType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:543
 PHPStan\Analyser\MutatingScope->getType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:1391
 PHPStan\Analyser\NodeScopeResolver->findEarlyTerminatingExpr() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:500
 PHPStan\Analyser\NodeScopeResolver->processStmtNode() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:302
 PHPStan\Analyser\NodeScopeResolver->processStmtNodes() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1306
 PHPStan\Analyser\MutatingScope->resolveType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:543
 PHPStan\Analyser\MutatingScope->getType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1391
 PHPStan\Analyser\MutatingScope->resolveType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:543
 PHPStan\Analyser\MutatingScope->getType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/ParametersAcceptorSelector.php:95
 PHPStan\Reflection\ParametersAcceptorSelector::selectFromArgs() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:3945
 PHPStan\Analyser\MutatingScope->methodCallReturnType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1854
 PHPStan\Analyser\MutatingScope->PHPStan\Analyser\{closure}() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1860
 PHPStan\Analyser\MutatingScope->resolveType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:543
 PHPStan\Analyser\MutatingScope->getType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1854
 PHPStan\Analyser\MutatingScope->PHPStan\Analyser\{closure}() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:1860
 PHPStan\Analyser\MutatingScope->resolveType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/MutatingScope.php:543
 PHPStan\Analyser\MutatingScope->getType() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:1361
 PHPStan\Analyser\NodeScopeResolver->findEarlyTerminatingExpr() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:500
 PHPStan\Analyser\NodeScopeResolver->processStmtNode() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:302
 PHPStan\Analyser\NodeScopeResolver->processStmtNodes() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:464
 PHPStan\Analyser\NodeScopeResolver->processStmtNode() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:302
 PHPStan\Analyser\NodeScopeResolver->processStmtNodes() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:548
 PHPStan\Analyser\NodeScopeResolver->processStmtNode() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:302
 PHPStan\Analyser\NodeScopeResolver->processStmtNodes() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:514
 PHPStan\Analyser\NodeScopeResolver->processStmtNode() at phar:///var/www/symfony/vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php:273
 PHPStan\Analyser\NodeScopeResolver->processNodes() at /var/www/symfony/vendor/rector/rector/packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:151
 Rector\NodeTypeResolver\PHPStan\Scope\PHPStanNodeScopeResolver->processNodesWithDependentFiles() at /var/www/symfony/vendor/rector/rector/packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:142
 Rector\NodeTypeResolver\PHPStan\Scope\PHPStanNodeScopeResolver->processNodes() at /var/www/symfony/vendor/rector/rector/packages/NodeTypeResolver/NodeScopeAndMetadataDecorator.php:63
 Rector\NodeTypeResolver\NodeScopeAndMetadataDecorator->decorateNodesFromFile() at /var/www/symfony/vendor/rector/rector/src/Application/FileProcessor.php:49
 Rector\Core\Application\FileProcessor->parseFileInfoToLocalCache() at /var/www/symfony/vendor/rector/rector/src/Application/FileProcessor/PhpFileProcessor.php:141
 Rector\Core\Application\FileProcessor\PhpFileProcessor->parseFileAndDecorateNodes() at /var/www/symfony/vendor/rector/rector/src/Application/FileProcessor/PhpFileProcessor.php:85
 Rector\Core\Application\FileProcessor\PhpFileProcessor->process() at /var/www/symfony/vendor/rector/rector/src/Application/ApplicationFileProcessor.php:175
 Rector\Core\Application\ApplicationFileProcessor->processFiles() at /var/www/symfony/vendor/rector/rector/src/Application/ApplicationFileProcessor.php:148
 Rector\Core\Application\ApplicationFileProcessor->run() at /var/www/symfony/vendor/rector/rector/src/Console/Command/ProcessCommand.php:147
 Rector\Core\Console\Command\ProcessCommand->execute() at /var/www/symfony/vendor/rector/rector/vendor/symfony/console/Command/Command.php:282
 RectorPrefix20220303\Symfony\Component\Console\Command\Command->run() at /var/www/symfony/vendor/rector/rector/vendor/symfony/console/Application.php:861
 RectorPrefix20220303\Symfony\Component\Console\Application->doRunCommand() at /var/www/symfony/vendor/rector/rector/vendor/symfony/console/Application.php:289
 RectorPrefix20220303\Symfony\Component\Console\Application->doRun() at /var/www/symfony/vendor/rector/rector/src/Console/ConsoleApplication.php:66
 Rector\Core\Console\ConsoleApplication->doRun() at /var/www/symfony/vendor/rector/rector/vendor/symfony/console/Application.php:187
 RectorPrefix20220303\Symfony\Component\Console\Application->run() at /var/www/symfony/vendor/rector/rector/bin/rector.php:57
 require_once() at /var/www/symfony/vendor/rector/rector/bin/rector:5

process [-n|--dry-run] [-a|--autoload-file AUTOLOAD-FILE] [--no-progress-bar] [--no-diffs] [--output-format OUTPUT-FORMAT] [--memory-limit MEMORY-LIMIT] [--clear-cache] [--port PORT] [--identifier IDENTIFIER] [--] [<source>...]

```

When I have specified copy of my `phpstan.neon` config file using the `Rector\Core\Configuration\Option::PHPSTAN_FOR_RECTOR_PATH` where I have disabled the `phpstan/phpstan-doctrine` extension, the error disappeared.

So I think I have found my workaround, but I wonder what will be the proper solution? Should this be somehow addressed by the `phpstan/phpstan-doctrine` extension (by adding some option which rector could leverage so that the prefixed version of symfony is used by the extension), or can this somehow be addressed directly by rector? :thinking: 

PS: Thx for all your great work, I bought your rector book. :ok_hand:

**Comment by dorrogeray** [comment]
This is probably to be expected, but I just found out that same problem manifests itself when using `phpstan/phpstan-symfony`, in particular when using the analysis of commands via:

```
    symfony:
        containerXmlPath: var/cache/dev/App_KernelDevDebugContainer.xml
        # This causes similar crashing
        consoleApplicationLoader: tests/console-application.php
```

Error when running rector with debug option: `php vendor/bin/rector process --clear-cache --debug`

...
...
src/Command/SomeCommand.php
Fatal error: Uncaught TypeError: Argument 1 passed to RectorPrefix20220323\Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator::__construct() must be an instance of RectorPrefix20220323\Symfony\Component\DependencyInjection\ContainerBuilder, instance of Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationContainerBuilder given, called in /var/www/symfony/vendor/symfony/dependency-injection/Loader/PhpFileLoader.php on line 67 and defined in /var/www/symfony/vendor/rector/rector/vendor/symfony/dependency-injection/Loader/Configurator/ContainerConfigurator.php:52
Stack trace:
#0 /var/www/symfony/vendor/symfony/dependency-injection/Loader/PhpFileLoader.php(67): RectorPrefix20220323\Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator->__construct(Object(Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationContainerBuilder), Object(Symfony\Component\DependencyInjection\Loader\PhpFileLoader), Array, '/var/www/symfon...', 'web.php', in /var/www/symfony/vendor/rector/rector/vendor/symfony/dependency-injection/Loader/Configurator/ContainerConfigurator.php on line 52

**Comment by TomasVotruba** [comment]
This might help to fix this soon:
https://github.com/rectorphp/rector-src/pull/2019

**Comment by samsonasik** [comment]
This should be resolved by using new RectorConfig on latest 0.12.21.

I am closing it.

**Comment by dorrogeray** [comment]
Hi again, sorry for the wall of text.

**TL;DR:**

**Bad news:** Unfortunately, the problem still persists
**Good news:** I have found relatively good workaround

**Workaround:**
1. Make sure that your DI container is compiled before rector runs
2. Make sure that the phpstan extensions never trigger DI container recompilation

For first point, I added this line to the top of my `rector.php` file:
```diff
+ require_once __DIR__ . '/compile-container.php';
```

The script `compile-container.php`:
```php
<?php

declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';

use Symfony\Component\Process\Exception\ProcessFailedException;
use Symfony\Component\Process\Process;

// Recompile the container if required
$process = new Process(['bin/console', 'about']);

$isDebug = in_array('--debug', $_SERVER['argv'], true);

if ($isDebug) {
    echo "Process [".getmypid()."] is making sure DI container is ready";
}

$process->run();

if (!$process->isSuccessful()) {
    echo " [ERROR]" . PHP_EOL;
    throw new ProcessFailedException($process);
}

if ($isDebug) {
    echo " [OK]".PHP_EOL;
}
```

For second point, you will have to make sure that your Kernel trusts the generated cache, which is a problem, because rector changes the files it iterates (the mtime changes). So I made sure that my Kernel trusts the cache without checking if the cache is fresh when the container is being initialized inside of a rector process:

```php
// inside src/Kernel.php

/**
 * Capture warmup dir
 *
 * {@inheritdoc}
 */
public function reboot(?string $warmupDir): void
{
    $this->warmupDir = $warmupDir;
    parent::reboot($warmupDir);
}

protected function initializeContainer(): void {
    $containerInitialized = false;
    // If rector is running, we don't have to check if the cache is fresh (we need to prevent container re-building,
    // otherwise phpstan extensions will break)
    if (defined('__RECTOR_RUNNING__')) {
        $isDebug = in_array('--debug', $_SERVER['argv'], true);
        $class = $this->getContainerClass();
        $buildDir = $this->warmupDir ?: $this->getBuildDir();
        $file = $buildDir.'/'.$class.'.php';

        if ($isDebug) {
            echo sprintf("[ Kernel ] Container is being initialized by Rector, cache will be considered fresh [pid=%d]\n", getmypid());
        }

        // Silence E_WARNING to ignore "include" failures - don't use "@" to prevent silencing fatal errors
        $errorLevel = error_reporting(\E_ALL ^ \E_WARNING);

        try {
            if (is_file($file) && \is_object($this->container = include $file)) {
                $this->container->set('kernel', $this);
                error_reporting($errorLevel);

                $containerInitialized = true;
            }
        } catch (\Throwable $throwable) {
        } finally {
            error_reporting($errorLevel);
        }
    }

    if (!$containerInitialized) {
        parent::initializeContainer();
    }
}
```

Thanks to these changes, both `phpstan/phpstan-doctrine` and `phpstan/phpstan-symfony` work correctly.

**Now the long version**, for those interested - why its necessary to prevent the container re-compilation inside rector when its leveraging phpstan extension which initializes the container:

First problem is that symfony sometimes includes multiple classes in the same file, like here: [MergeExtensionConfigurationPass.php](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php)

Because classes A, B and C are declared in same file, and class B is used inside of class A, it is not possible to create class alias for classes B and C, because once class A is autoloaded (when creating alias for class A), the classes B and C get loaded as well (because they are in the same file) and after that its no longer possible to declare the alias, because the class is "already in use".

I have tried "fixing" this problem locally, so I split the three classes into three files in symfony, and also in the prefixed rector vendor directory. This fixed the problem with autoloading, but another problem popped up right after that.

When the rector, through phpstan, tries to access one of the phpstan extensions (for example, `phpstan/phpstan-doctrine`), and this extensions tries to boots its kernel, the compilation of the container crashes. Extensions like [FrameworkExtension](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php) which are necessary for the compilation are not supported by rector (there is no prefixed version of this class). Basically, from that point, the unprefixed version of `FrameworkExtension` starts working with other prefixed classes (the aliasing has already happened at this point), and then it just crashes (if I would fix the next error, more would pop up I am sure). I did notice that there is prefixed interface for the [ExtensionInterface](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/DependencyInjection/Extension/ExtensionInterface.php), but on its own thats not enough to make the container compilation work.

I am not saying that rector should (or should not) support full compilation of the symfony container, just letting you know that this is the reason why these phpstan extensions are not playing along with rector.

For example, if you would decide to add prefixed FrameworkBundle to rector, there would probably be similar problem with other bundles.. And it does not seem feasible to include them all (or in all versions).. Only option would be to generate the prefixed vendor directory "on demand", but that seems over-engineered..

**Comment by TomasVotruba** [comment]
Hi, thanks for sharing the approach and setup for your project.
Let's keep this issue open until the bad news is solved :) 

The solution has 3 steps:

* introduce a custom config class independent on Symfony :heavy_check_mark: 
* make users use config with new `RectorConfig` :heavy_check_mark:  (WIP :))
* start prefixing the Symfony container class to avoid this collission - to be done

**Comment by TomasVotruba** [comment]
The `RectorConfig` is now solved and migration is proposed on every Rector run.
In next version this will throw an exceptoin to inform users about migration.

Next steps is to prefix the `ContainerConfigurator` of Symfony, as proposed in https://github.com/rectorphp/rector-src/pull/2104

**Comment by TomasVotruba** [comment]
We've upgraded the PHP Scoper from 0.14 to 0.17, there the "unprefixing" mechanism changed, so it might require some hacking around.
Could you share mininal reproducer so we can see what is the minimal conditions that cause it?
